IGNITE-16021 Update Apache Ignite 2.12 release notes (#9620)
apache/ignite: RELEASE_NOTES.txt

Apache Ignite Release Notes
===========================

Apache Ignite In-Memory Distributed Database 2.12.0
-----------------------------------------------------------

(!) WARNINGS:

> * The community accepted deprecate for removal in future releases: CacheMode#LOCAL, CacheAtomicityMode#TRANSACTIONAL_SNAPSHOT, CacheConfiguration#rebalanceDelay.

Reviewer: We should mention that legacy service grid implementation is already removed in the next relese.

Author: Fixed.

* Added batch cache operations histogram metrics.
* Added benchmarks for cache queries - scan, index, text.
* Added docker image for s390x.
* Added events for snapshot restore operation.
* Added expire policy info into log of started cache.
* Added offline utility to read and analyze index files.
* Added possibility to accept while indexing classes/enums that are extending interfaces/classes marked to be stored in cache.
* Added rename index tree operation with corresponding WAL record.
* Added resource de-allocation in SharedPageLockTracker.
* Added support for creating IndexQuery without index name.
* Added the ability to cancellation of consistency recovery command (Read Repair via control.ch).
* Added the ability to record Control.sh consistency check violations to other log file.
* Added the ability to restore snapshot taken on different topologies.
* Added the ability to snapshot encrypted caches.
* Added the cache destroy command for control.sh.
* Added the force deactivation flag to the IgniteMXBean.
* Added time metrics and statistics for the IgniteCache#getAllOutTx.
* Added unconditional logging of tx states to WAL  to ensure correct tx recovery after node crash.

> * Any readRepair attempt now generates the event (if consistency violation found), even for atomic caches.

Reviewer: ```suggestion
* Added firing event for any ReadRepair attempt (if consistency violation found). Atomic caches are also supported.
```

Author: Fixed.

* Updated PostgreSQL JDBC Driver version.
* Updated httpclient, httpcore versions.
* Updated the Jackson dependency version.
* Updated the MySql connector dependency version.
* Updated the Netty dependency version.

Java thin-client:
* Added SQLSTATE to thin client SQL error message.
* Added client cache for OptimizedMarshaller class names.
* Added partition awareness for ScanQuery with specified partition.
* Added requests thread pool monitoring.
* Fixed ClassNotFoundException on service call after failover.
* Fixed follow user-defined endpoint order, try default port first.
* Fixed issue with explicit binary type configuration.
* Fixed transaction failure after timeout.

.Net:
* Fixed NRE in ClientFailoverSocket due to late logger setter.
* Fixed NullPointerException in ContinuousQuery with security enabled.
* Fixed Schema project version and examples packaging.
* Fixed SslStreamFactory.CertificatePath null value.
* Fixed TypeNameParser to ignore escaped characters in compiler-generated type names.
* Fixed dynamic assemblies handling in TypeResolver.
* Fixed thin client streamer not creating SQL table entries.
* Fixed verify-nuget.ps1 failure when .NET 5 is installed.

> * Modernized project structure.

Reviewer: I don't think we should mention it in the release notes.

Author: Fixed.
